frontend/src/scenes/billing/BillingHero.tsx

```diff
@@ -1,26 +1,220 @@
-import './BillingHero.scss'
+import { LemonButton } from '@posthog/lemon-ui'
+import { Link } from '@posthog/lemon-ui'
+import { useActions, useValues } from 'kea'
+import { BillingUpgradeCTA } from 'lib/components/BillingUpgradeCTA'
+import { FEATURE_FLAGS } from 'lib/constants'
+import { featureFlagLogic } from 'lib/logic/featureFlagLogic'
+import planEnterprise from 'public/plan_enterprise.svg'
+import planFree from 'public/plan_free.svg'
+import planPaid from 'public/plan_paid.svg'
+import planStartup from 'public/plan_startup.svg'
+import planYc from 'public/plan_yc.svg'
 
-import { BlushingHog } from 'lib/components/hedgehogs'
-import useResizeObserver from 'use-resize-observer'
+import { BillingPlan, BillingProductV2Type, StartupProgramLabel } from '~/types'
 
-export const BillingHero = (): JSX.Element => {
-    const { width, ref: billingHeroRef } = useResizeObserver()
+import { getUpgradeProductLink } from './billing-utils'
+import { billingLogic } from './billingLogic'
+import { billingProductLogic } from './billingProductLogic'
+import { paymentEntryLogic } from './paymentEntryLogic'
+import { PlanComparisonModal } from './PlanComparison'
+
+const PLAN_BADGES: Record<BillingPlan, string> = {
+    [BillingPlan.Free]: planFree,
+    [BillingPlan.Paid]: planPaid,
+    // TODO: Add teams badge once ready
+    [BillingPlan.Teams]: planPaid,
+    [BillingPlan.Enterprise]: planEnterprise,
+}
+
+const STARTUP_PROGRAM_BADGES: Record<StartupProgramLabel, string> = {
+    [StartupProgramLabel.YC]: planYc,
+    [StartupProgramLabel.Startup]: planStartup,
+}
+
+interface CopyVariation {
+    title: string | null
+    subtitle: string | null
+    getDescription: (billingPlan: BillingPlan, scrollToProduct: (productType: string) => void) => JSX.Element
+    backgroundColor: string
+}
+
+const BADGE_CONFIG: Record<BillingPlan | StartupProgramLabel, CopyVariation> = {
+    [BillingPlan.Free]: {
+        title: 'Get the whole hog.',
+        subtitle: 'Only pay for what you use.',
+        backgroundColor: 'bg-danger-highlight',
+        getDescription: () => (
+            <>
+                <p>PostHog comes with all product features on every plan.</p>
+                <p>Add your credit card to remove usage limits and unlock all platform features.</p>
+                <p>Set billing limits as low as $0 to control your spend.</p>
+                <p className="italic">P.S. You still keep the monthly free allotment for every product!</p>
```

Comment:
```suggestion
                <p className="italic">P.S. You still keep the monthly free allowance for every product!</p>
```

I think allowance is a slightly preferred term